test/lib/random_schema.cc

```diff
@@ -803,7 +803,7 @@ expiry_generator no_expiry_expiry_generator() {
 
 namespace {
 
-schema_ptr build_random_schema(uint32_t seed, random_schema_specification& spec) {
+schema_ptr build_random_schema(uint32_t seed, random_schema_specification& spec, bool compress) {
```

Comment:
Better make the new parameter part of random_schema_specification rather than add a new way to influence the schema. It can default to the current behavior.